hack/tests/Makefile

```diff
@@ -0,0 +1,202 @@
+.docker-build: .docker-build-capi .docker-build-capk
+
+.generate-capi-manifests: .ensure-capi-path
+	@make -C $(CAPI_PATH) generate-manifests REGISTRY=$(DEV_REGISTRY) PULL_POLICY=Always
+	@make -C ${CAPI_PATH}/test/infrastructure/docker generate-manifests REGISTRY=$(DEV_REGISTRY) PULL_POLICY=Always
+
+.generate-capk-manifests:
+	@make -C $(ROOT_DIR) generate-manifests REGISTRY=$(DEV_REGISTRY) PULL_POLICY=Always
+
+.generate-manifests: .generate-capi-manifests .generate-capk-manifests
+
+CLUSTERCTL_CONFIG_PATH := $(HOME)/.cluster-api
+.create-local-repository: .ensure-capi-path
+	@rm -fr $(CLUSTERCTL_CONFIG_PATH)
+	@cd $(CAPI_PATH) && PATH=$${PATH}:$(abspath $(CAPI_PATH)) && ./cmd/clusterctl/hack/create-local-repository.py
+
+
+define LOCAL_REPOSITORY_PATCH
+cert-manager:
+  url: "$(ROOT_DIR)/hack/tests/resources/cert-manager/v1.1.0/cert-manager.yaml"
+images:
+  all:
+    repository: "$(DEV_REGISTRY)"
+endef
+
+
+.install-cert-manager:
+	@docker pull $(CERT_MANAGER_IMAGE_REPOSITORY)/$(CERT_MANAGER_CAINJECTOR):$(CERT_MANAGER_VERSION)
+	@docker tag $(CERT_MANAGER_IMAGE_REPOSITORY)/$(CERT_MANAGER_CAINJECTOR):$(CERT_MANAGER_VERSION) $(DEV_REGISTRY)/$(CERT_MANAGER_CAINJECTOR):$(CERT_MANAGER_VERSION)
+	@docker push $(DEV_REGISTRY)/$(CERT_MANAGER_CAINJECTOR):$(CERT_MANAGER_VERSION)
+
+	@docker pull $(CERT_MANAGER_IMAGE_REPOSITORY)/$(CERT_MANAGER_CONTROLLER):$(CERT_MANAGER_VERSION)
+	@docker tag $(CERT_MANAGER_IMAGE_REPOSITORY)/$(CERT_MANAGER_CONTROLLER):$(CERT_MANAGER_VERSION) $(DEV_REGISTRY)/$(CERT_MANAGER_CONTROLLER):$(CERT_MANAGER_VERSION)
+	@docker push $(DEV_REGISTRY)/$(CERT_MANAGER_CONTROLLER):$(CERT_MANAGER_VERSION)
+
+	@docker pull $(CERT_MANAGER_IMAGE_REPOSITORY)/$(CERT_MANAGER_WEBHOOK):$(CERT_MANAGER_VERSION)
+	@docker tag $(CERT_MANAGER_IMAGE_REPOSITORY)/$(CERT_MANAGER_WEBHOOK):$(CERT_MANAGER_VERSION) $(DEV_REGISTRY)/$(CERT_MANAGER_WEBHOOK):$(CERT_MANAGER_VERSION)
+	@docker push $(DEV_REGISTRY)/$(CERT_MANAGER_WEBHOOK):$(CERT_MANAGER_VERSION)
+
+	@docker pull $(KUBE_RBAC_IMAGE_REPOSITORY)/$(KUBE_RBAC_PROXY):$(KUBE_RBAC_PROXY_VERSION)
+	@docker tag $(KUBE_RBAC_IMAGE_REPOSITORY)/$(KUBE_RBAC_PROXY):$(KUBE_RBAC_PROXY_VERSION) $(DEV_REGISTRY)/$(KUBE_RBAC_PROXY):$(KUBE_RBAC_PROXY_VERSION)
+	@docker push $(DEV_REGISTRY)/$(KUBE_RBAC_PROXY):$(KUBE_RBAC_PROXY_VERSION)
+
+.config-local-repository: export LOCAL_REPOSITORY_PATCH:=$(LOCAL_REPOSITORY_PATCH)
+.config-local-repository: $(CLUSTERCTL_CONFIG_PATH)
+	@echo "$${LOCAL_REPOSITORY_PATCH}" >> $(CLUSTERCTL_CONFIG_PATH)/dev-repository/config.yaml
+
+## --------------------------------------
+## Deploy CAPI
+## --------------------------------------
+
+CLUSTERCTL_BIN := $(abspath $(CAPI_PATH)/bin/)
+$(CLUSTERCTL_BIN): .ensure-capi-path
+	@make -C $(CAPI_PATH) clusterctl
+
+.deploy-cluster-api:
+	$(CAPI_PATH)/bin/clusterctl init \
+	 --core cluster-api:v0.4.99 \
+	 --bootstrap kubeadm:v0.4.99 \
+	 --control-plane kubeadm:v0.4.99 \
+	 --infrastructure kubemark:v0.2.0 \
+	 --infrastructure docker:v0.4.99 \
```

Comment:
@elmiko - I think this can be done at a point when we declare compatibility with a newer version of capi. i.e. when we upgrade capi, and run this CI script, it will fail ... so then we also update the CI script. by that, effectively declaring that we are now compatible with the newer capi version

(at the same time also update the versions on the README)